cloudify/utils.py

```diff
@@ -1058,3 +1060,17 @@ def uuid4():
         uuid_as_hex[16:20],
         uuid_as_hex[20:]
     )
+
+
+def keep_trying_http(func, total_timeout_sec=None):
```

Comment:
Should we default to no timeout?

Reply:
I think we should, at least for agents.  They probably shouldn't be allowed to perform any new operations until the current one is finished and successfully reported as such.
Of course, your question is more interesting in a broader scope.  Maybe we would wrap some more requests (not just `CloudifyContext.update_operation()`) with `keep_trying_http`.  I'll take a look.

Reply:
Not performing new operations probably makes sense (will they anyway?), but they should probably be able to error out at some point.

Also, I note there's no backoff here, so if the server is being overwhelmed by traffic then they'll just hammer it as fast as they can.
There's also no logging so one can't tell when a problem is occurring by looking anywhere.